core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/transferprocess/TransferProcessServiceImpl.java

```diff
@@ -126,7 +126,7 @@ public ServiceResult<Stream<TransferProcess>> query(QuerySpec query) {
     public @NotNull ServiceResult<String> initiateTransfer(DataRequest request, ClaimToken claimToken) {
         return transactionContext.execute(() ->
                 Optional.ofNullable(negotiationStore.findContractAgreement(request.getContractId()))
-                        .filter(agreement -> contractValidationService.validateAgreement(claimToken, agreement))
+                        .filter(agreement -> contractValidationService.validateAgreement(claimToken, agreement).succeeded())
```

Comment:
This is ok, but the failed result will be not reported to the caller. Anyway I think that we could address this to another issue.